clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleLoadBalancerCachingAgent.groovy

```diff
@@ -0,0 +1,145 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.google.provider.agent
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.google.api.client.googleapis.batch.BatchRequest
+import com.google.api.client.googleapis.batch.json.JsonBatchCallback
+import com.google.api.client.googleapis.json.GoogleJsonError
+import com.google.api.client.http.HttpHeaders
+import com.google.api.services.compute.Compute
+import com.google.api.services.compute.model.ForwardingRule
+import com.netflix.spinnaker.cats.agent.AccountAware
+import com.netflix.spinnaker.cats.agent.AgentDataType
+import com.netflix.spinnaker.cats.agent.CacheResult
+import com.netflix.spinnaker.cats.agent.CachingAgent
+import com.netflix.spinnaker.cats.agent.DefaultCacheResult
+import com.netflix.spinnaker.cats.cache.CacheData
+import com.netflix.spinnaker.cats.cache.DefaultCacheData
+import com.netflix.spinnaker.cats.provider.ProviderCache
+import com.netflix.spinnaker.clouddriver.google.GoogleCloudProvider
+import com.netflix.spinnaker.clouddriver.google.cache.Keys
+import com.netflix.spinnaker.clouddriver.google.model.GoogleLoadBalancer
+import com.netflix.spinnaker.clouddriver.google.model.GoogleResourceRetriever
+import com.netflix.spinnaker.clouddriver.google.model.callbacks.Utils
+import com.netflix.spinnaker.clouddriver.google.provider.GoogleInfrastructureProvider
+import groovy.transform.TupleConstructor
+import groovy.util.logging.Slf4j
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
+
+@Slf4j
+class GoogleLoadBalancerCachingAgent implements CachingAgent, AccountAware {
+
+  // TODO(ttomsu): This will be shared across a few caching agents. Move to common class.
+  static final TypeReference<Map<String, Object>> ATTRIBUTES = new TypeReference<Map<String, Object>>() {}
+
+  final GoogleCloudProvider googleCloudProvider
+  final String googleApplicationName // "Spinnaker/${version}" HTTP header string
+  final String accountName
+  final String region
+  final String project
+  final Compute compute
+  final ObjectMapper objectMapper
+
+  final String providerName = GoogleInfrastructureProvider.PROVIDER_NAME
+  final Set<AgentDataType> providedDataTypes = Collections.unmodifiableSet([
+    AUTHORITATIVE.forType(Keys.Namespace.LOAD_BALANCERS.ns)
```

Comment:
Yep, that's the idea. Haven't gotten to the server group part yet, though.
I wanted to make sure I was on the right track
On Feb 8, 2016 6:16 PM, "Lars Wander" <notifications@github.com> wrote:

> In
> clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleLoadBalancerCachingAgent.groovy
> <https://github.com/spinnaker/clouddriver/pull/288#discussion_r52243867>:
>
> > +class GoogleLoadBalancerCachingAgent implements CachingAgent, AccountAware {
> > +
> > +  // TODO(ttomsu): This will be shared across a few caching agents. Move to common class.
> > +  static final TypeReference<Map<String, Object>> ATTRIBUTES = new TypeReference<Map<String, Object>>() {}
> > +
> > +  final GoogleCloudProvider googleCloudProvider
> > +  final String googleApplicationName // "Spinnaker/${version}" HTTP header string
> > +  final String accountName
> > +  final String region
> > +  final String project
> > +  final Compute compute
> > +  final ObjectMapper objectMapper
> > +
> > +  final String providerName = GoogleInfrastructureProvider.PROVIDER_NAME
> > +  final Set<AgentDataType> providedDataTypes = Collections.unmodifiableSet([
> > +    AUTHORITATIVE.forType(Keys.Namespace.LOAD_BALANCERS.ns)
>
> Won't you be updating relationships for instances/server groups or
> whatever you wind up associating with load balancers?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spinnaker/clouddriver/pull/288/files#r52243867>.
>